### PR TITLE
API v2 Storefront Checkout finalize

### PR DESCRIPTION
**Pull request by damianlegawiec**
- [X] Added Checkout Shipping Rates endpoint (`/checkout/shipping_ratea`)
- [X] Fixed `ShippingRate#tax_amount` method if no tax is applied
- [X] Added specs for selecting Shipping Rates
- [X] Added request specs for full checkout flow
- [X] Updated Checkout OpenAPI docs

### CODE REVIEW

File: api/app/serializers/spree/v2/storefront/shipment_serializer.rb [hunk 1]
```diff
@@ -4,8 +4,12 @@ module Storefront
...
         set_type :shipment
 
-        attributes :tracking, :number, :cost, :display_cost, :discounted_cost, :display_discounted_cost,
-                   :final_price, :display_final_price, :item_cost, :display_item_cost, :shipped_at
+        attributes :number, :final_price, :display_final_price,
+                   :state, :shipped_at, :tracking_url
+
+        attribute :free, &:free?
+
+        has_many :shipping_rates, if: proc { |_record, params| params && params[:show_rates] == true }
```
Comment (piotrleniec-spark) [inline api/app/serializers/spree/v2/storefront/shipment_serializer.rb]: Maybe we could write `params&.dig(:show_rates)` or `params&.fetch(:show_rates, false)` here?

File: core/app/models/spree/shipment.rb [hunk 2]
```diff
@@ -134,6 +135,10 @@ def final_price_with_items
       item_cost + final_price
     end
 
+    def free?
+      final_price == BigDecimal(0) || adjustments.promotion.any? { |p| p.source.type == 'Spree::Promotion::Actions::FreeShipping' }
```
Comment (piotrleniec-spark) [inline core/app/models/spree/shipment.rb]: Maybe we could use a guard clause to split this long condition into multiple lines?

**Review by pl-spark — APPROVED** [review]

**Review by przemosk — APPROVED** [review]

**Review by btolarz — APPROVED** [review]

File: core/app/models/spree/shipment.rb (hunk 2 above)
Comment (damianlegawiec) [reply core/app/models/spree/shipment.rb]: done!

File: api/app/serializers/spree/v2/storefront/shipment_serializer.rb (hunk 1 above)
Comment (damianlegawiec) [reply api/app/serializers/spree/v2/storefront/shipment_serializer.rb]: fixed!

**Pull request merged**